### PR TITLE
Move CRDs to yaml

### LINKED ISSUE #736: Create NSM CRD bundle yaml
Today NSM registers its CRDs in NSMD-k8s here: https://github.com/networkservicemesh/networkservicemesh/blob/51295d0cd3bfdbe8c3a99e05ad8974d19c6d9a3a/k8s/cmd/nsmd-k8s/main.go#L112

They are removed when NSMD-k8s is killed here:
https://github.com/networkservicemesh/networkservicemesh/blob/51295d0cd3bfdbe8c3a99e05ad8974d19c6d9a3a/k8s/cmd/nsmd-k8s/main.go#L134

This add/remove procedure is not as robust as we would like. The proposed (and established) way of registering the project CRDs would be to have a dedicated yaml bundle that describes them.

These would become part of our cluster configuration, as we do with the RBACs here:
https://github.com/networkservicemesh/networkservicemesh/blob/master/.k8s.mk#L29

We also want NSMD-k8s code to deal with CRDs add/remove to be deleted.

Note that the initializing the Packet cluster for the CI needs to be adapted for your CI to pass. It is done here:
https://github.com/networkservicemesh/networkservicemesh/blob/master/scripts/prepare-circle-integration-tests.sh#L31

### PR DESCRIPTION
**Pull request by ivanayov**
This creates NSM CRD bundle yaml and removes CRD generation and cleanup from code

Signed-off-by: Ivana Yovcheva <iyovcheva@vmware.com>

## Motivation and Context

Closes #736

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

```
$ make k8s-icmp-delete
$ make k8s-vpn-delete

$ make k8s-save

$ make k8s-infra-deploy

$ make k8s-icmp-deploy
$ make k8s-vpn-deploy
$ make k8s-check
```
Check CRDs are updated
```
$ kubectl get crd
NAME                                            CREATED AT
networkserviceendpoints.networkservicemesh.io   2019-02-13T12:18:43Z
networkservicemanagers.networkservicemesh.io    2019-02-13T12:18:44Z
networkservices.networkservicemesh.io           2019-02-13T12:18:43Z
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .k8s.mk [hunk 1]
```diff
@@ -26,7 +26,7 @@ DEPLOY_ICMP = $(DEPLOY_ICMP_KERNEL) $(DEPLOY_ICMP_VPP)
 DEPLOY_VPN = secure-intranet-connectivity vppagent-firewall-nse vpn-gateway-nse vpn-gateway-nsc
 DEPLOYS = $(DEPLOY_INFRA) $(DEPLOY_ICMP) $(DEPLOY_VPN)
 
-CLUSTER_CONFIGS = cluster-role-admin cluster-role-binding cluster-role-view
+CLUSTER_CONFIGS = cluster-role-admin cluster-role-binding cluster-role-view crd-networkservices crd-networkserviceendpoints crd-networkservicemanagers
```
Comment (nickolaev) [inline .k8s.mk]: I'd split these into `CLUSTER_CONFIG_ROLE` an `CLUSTER_CONFIG_CRD` and then `CLUSTER_CONFIG = $(CLUSTER_CONFIG_ROLE) $(CLUSTER_CONFIG_CRD)` Makes thisngs more manageable.

File: docs/QUICK-START.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ make k8s-infra-deploy
 A simple check should show two `nsmd`, two `nsm-vppagent-dataplane`, two `skydive-agent`, one `crossconnect-monitor` and one `skydive-analyzer` pods.
 
 ```bash
-kubect get pods
+kubectl get pods
```
Comment (nickolaev) [inline docs/QUICK-START.md]: Good catch, thanks.

File: docs/QUICK-START.md [hunk 3]
```diff
@@ -134,6 +134,7 @@ NSC nsc-84799b768f-wvxwt with IP 10.30.1.1/30 pinging vppagent-icmp-responder-ns
 Validating the VPN example will output the following:
 
 ```bash
+./scripts/verify_vpn_gateway.sh
```
Comment (nickolaev) [inline docs/QUICK-START.md]: Actually this script is called by `make k8s-check`, I don't think we need it here.

File: k8s/cmd/nsmd-k8s/main.go [hunk 4]
```diff
@@ -75,69 +59,6 @@ func main() {
...
-	signal.Notify(c, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
-	go func() {
-		<-c
-		removeCRDs(clientset)
-	}()
-
 	err = server.Serve(listener)
 	logrus.Fatalln(err)
 }
-
```
Comment (nickolaev) [inline k8s/cmd/nsmd-k8s/main.go]: I like the cleanup.

File: k8s/conf/crd-networkserviceendpoints.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  - lastTransitionTime: null
+    reason: NoConflicts
+    status: "True"
+    type: NamesAccepted
+  - lastTransitionTime: null
+    reason: InitialNamesAccepted
+    status: "True"
+    type: Established
+  storedVersions:
+  - v1
```
Comment (nickolaev) [inline k8s/conf/crd-networkserviceendpoints.yaml]: New line here please.

File: k8s/conf/crd-networkservicemanagers.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  - lastTransitionTime: null
+    reason: NoConflicts
+    status: "True"
+    type: NamesAccepted
+  - lastTransitionTime: null
+    reason: InitialNamesAccepted
+    status: "True"
+    type: Established
+  storedVersions:
+  - v1
```
Comment (nickolaev) [inline k8s/conf/crd-networkservicemanagers.yaml]: New line here please.

File: k8s/conf/crd-networkservices.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: networkservices.networkservicemesh.io
+  # selfLink: /apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/networkservices.networkservicemesh.io
```
Comment (nickolaev) [inline k8s/conf/crd-networkservices.yaml]: Is this serving a purpose? If not, please remove it.

**Comment by nickolaev** [comment]
@ivanayov you can try `make k8s-integration-TestSeveralCrossConnects-test` on your host and see kubectl logs why the `nsc1` does not come up. It may have something to do with the CRD definitions(yaml) not being complete. You can check `nsc1` intiCOntainer logs. Somehting like`kubectl logs nsc1 nsc`.

**Comment by ivanayov** [comment]
@nickolaev `make k8s-integration-tests` has now no failures locally

**Pull request merged**